subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/VersionedNativeCompiler.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.nativeplatform.toolchain.internal;
+
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Internal;
+import org.gradle.util.VersionNumber;
+
+public interface VersionedNativeCompiler {
+    @Internal
+    VersionNumber getVersion();
+    @Input
+    String getVersionString();
+    @Input
+    String getType();
+}
```

Comment:
I think this data should be captured in public `NativeCompilerVersion` class, and `VersionedNativeCompiler` should only expose a single method returning that object. We could go more generic and have a `VersionedCompiler` interface returning a `CompilerVersion` object.

`VersionedNativeCompiler` should extend `Compiler`. If we do `VersionedNativeCompiler`, it would be nice to extract `NativeCompiler` as an interface, too.

Another approach would be to introduce `VersionedTool` independent of the `Compiler` hierarchy, and have it return a `ToolVersion` object that we could track.

Reply:
I went down the `NativeCompilerVersion` route.